File: javaa/Manusia.java
package javaa;

public class Manusia {
    private String nama;
    private int umur; 
    private String gender;

    public Manusia(String nama, int umur, String gender) {
        //pakai setter biar langsung satset SRP
        setNama(nama);
        setUmur(umur);
        setGender(gender);
    }

    // getter
    public String getNama() {
        return nama;
    }
    public int getUmur() {
        return umur;
    }
    public String getGender() {
        return gender;
    }

    // setter
    public void setNama(String nama) {
        if (nama == "") {
            System.out.println("Nama tidak boleh kosong");
            return; //exit
        } 
        this.nama = nama;
    }
    public void setUmur(int umur) {
        if (umur <= 0) {
            System.out.println("Umur tidak valid");
            return; //exit
        }
        this.umur = umur;
    }
    public void setGender(String gender) {
        this.gender = gender;
    }

    //method
    public void informasi() {
        System.out.println(getNama());
        System.out.println(getUmur());
        System.out.println(getGender());
    }
}

File: pythonn/Manusia.py


class Manusia :
    def __init__(self, nama, umur, gender) : 
        self.setNama(nama) #harus pakai prefix self kalau scopenya sama satu class
        self.setUmur(umur)
        self.gender = gender

    #setter
    def setNama(self, nama) :
        if (nama == "") : 
            print("Nama tidak boleh kosong wok")
            return #early exit
        self.nama = nama

    def setUmur(self, umur) :
        if (umur <= 0) :
            print("Masukkan umur dengan benar")
            return 
        self.umur = umur

    def setGender(self, gender) :
        self.gender = gender

    #getter
    def getNama(self) :
        return self.nama

    def getUmur(self) :
        return self.umur

    def getGender(self) :
        return self.gender

    def informasi(self) :
        print(self.getNama())
        print(self.getUmur())
        print(self.getGender())